### ISSUE TITLE
English language issues during registration

### ISSUE THREAD

**Issue by Flameborn** [recoilme/tgram#60]
Hello,

I noticed  you have reverted changes in the _views/register.html_ template, mainly because of the joke checkbox.

Unfortunately, this joke is not really understandable by English-speaking people. I think I see what you mean here, but _good internet person_ does not have much meaning in English. Maybe _I am a well-behaved Netizen_? This would make a lot more sense.

Also, regarding the checkboxes to do with the ToS and Policy, _I am read and accept_ is not correct, it should be _I have read and accept the_.

**Comment by recoilme** [comment]
oh, thank you - i change this back(

**Comment by Flameborn** [comment]
Thank you.

Actually, you could have a checkbox and call it something like _I am here to spam and destroy the world_, or something similar, and raise an error if it is checked. The error could say something like _Sorry, this site is not dedicated to spamming and world domination, so we cannot accept your registration._

Apart from being a joke, which it is, it could also keep out spammers, because bots usually just check everything on a form and then submit.

This could be an additional security step besides recaptcha. Just a thought, what do you think?

**Comment by recoilme** [comment]
This very fun and useful ))
Maybe you implement your idea?

**Comment by Flameborn** [comment]
I'm glad you like it. Sure, I can do this.

I also noticed there are a few things that probably should be moved to the .env config file, such as the recaptcha site key, the name of the admin who can delete posts, the site name and description, etc. I'll see if I can do something about this as well, a bit later.

**Comment by recoilme** [comment]
nice idea about admin and so on
but I remove recaptcha and all code about
recaptcha works not stable and hated by people

**Comment by Flameborn** [comment]
There are a few problems with Recaptcha:

* It is bad if you worried about your privacy (even more control for Google)
* It is not reliable, as far as the V2 API is concerned. V3 is in beta now, and apparently it is unseen. But I can confirm that V2 on a self-hosted URL shortener got hundreds of spam urls every day, and Recaptcha was not able to stop it.

The best we can do is either email confirmation (which is not ideal if you do not have a mail server), or some sort of honeypot, e.g. a question like _What is the third planet from the sun?_ and JS code that fills in the answer for you, or if JS is disabled, you can fill it in manually. Unless you are targeted, spam bots won't know that the answer is _Earth_.

**Comment by recoilme** [comment]
i like I am here to spam and destroy the world, or something similar, and raise an error if it is checked

and have protection by ip

**Comment by Flameborn** [comment]
Yeah, I think it is a nice thing to implement as well. I'll do it.

**Comment by recoilme** [comment]
thank you!
